lib/src/Uncertainty/Bayesian/MetropolisHastingsImplementation.cxx

```diff
@@ -69,6 +71,8 @@ MetropolisHastingsImplementation::MetropolisHastingsImplementation(const Functio
   , initialState_(initialState)
   , currentState_(initialState)
   , marginalIndices_(marginalIndices)
```

Comment:
```suggestion
```
This line is useless (`setMarginalIndices` is called by the constructor in its body) and might be causing the warnings (treated as errors by CI machines).